website/api/recursion.md

```diff
@@ -0,0 +1,150 @@
+1. Each _Segment_ is proven, resulting in a _SegmentReceipt_.
+1. Each _SegmentReceipt_ is _lifted_, resulting in a _SuccinctReceipt_.
+1. Pairs of _SuccinctReceipts_ are _joined_, resulting in another _SuccinctReceipt_. This continues until a single _SuccinctReceipt_ remains.
+1. The final _SuccinctReceipt_ is passed through _identity_p254_, which prepares for Groth16 proving.
+1. The _SuccinctReceipt_ is _compressed_, generating a _Groth16Receipt_.
+
+The _Groth16Receipt_ can now be posted on-chain and verified by the [RISC Zero Verifier Contract].
+
+## Recursive Circuit Architecture
+
+RISC Zero's zkVM consists of three circuits.
+
+1. The RISC-V Circuit is a STARK circuit that proves correct execution of RISC-V programs.
+1. The Recursion Circuit is a separate STARK circuit, that's designed to efficiently generate proofs for the verification of STARK proofs and to support the integration of custom accelerator circuits into the zkVM.
+   This circuit has a similar architecture to the RISC-V Circuit, but with fewer columns and a much simpler instruction set.
+   The same [proof system] is used for both the RISC-V Circuit and the Recursion Circuit.
+1. The STARK-to-SNARK Circuit is an R1CS circuit that proves verification of proofs from the Recursion Circuit.
+
+## Recursion Programs
+
+The Recursion Circuit supports a number of programs, including `lift()`, `join()`, `resolve()`, and `identity_p254()`.
+Instead, users should access recursion via the [Prover].
+
+To aid in conceptual understanding, we offer the following explanations of the recursion programs:
+
+1. The `lift()` program verifies a STARK proof from the RISC-V Prover, using the Recursion Prover. This recursion proof has a single constant-time verification procedure, with respect to the original segment length, and is then used as the input to all other recursion programs (e.g. join, resolve, and identity_p254).
+
+1. The `join()` program verifies two proofs from the Recursion Prover, using the Recursion Prover. By repeated application of `join()`, any number of receipts for execution spans within the same session can be compressed into a single receipt for the entire session.
+
+1. The `identity_p254()` program verifies a proof from the Recursion Prover using the Poseidon254 hash function. The identity_p254 program is used as the last step in the prover pipeline before running the Groth16 prover.
+
+## STARK-to-SNARK Wrapping
+
+All of the recursion programs in the previous section output a [SuccinctReceipt], which is a STARK proof (~200kB)
+
+The final step in the recursion process is `compress()`, which outputs a [Groth16Receipt], which can be verified on-chain using the [RISC Zero Verifier Contract].
+
+[composite, succinct or groth16 receipts]: https://docs.rs/risc0-zkvm/1.0/risc0_zkvm/enum.ReceiptKind.html
+[Groth16Receipt]: https://docs.rs/risc0-zkvm/latest/risc0_zkvm/struct.Groth16Receipt.html
+[proof system]: /docs/proof-system/proof-system-sequence-diagram
```

Comment:
```suggestion
[proof system]: /docs/proof-system/proof-system-sequence-diagram.md
```

Remember #1972 :wink:

Reply:
@pdg744 you forgot to include changes from main that updated yarn - you must run `yarn` to install docusaurus 3.4.0 locally . Just merged main for you.

Strangely I cannot for the life of me resolve the right link :face_exhaling:  any ideas?

Reply:
with .md I CANNOT get anything to work. Without md, any path "works" but does not show up

![image](https://github.com/risc0/risc0/assets/35669742/1d51ab40-8d85-4e80-9bc2-c6b962464b89)

![Uploading image.png…]()

Reply:
This works locally for me now, without the .md - but not for the "right" reason with the link checker: 

![image](https://github.com/risc0/risc0/assets/35669742/1214a4be-4276-47af-89dd-aab6a594efe1)